File: 789-kth-largest-element-in-a-stream/kth-largest-element-in-a-stream.cpp
class KthLargest {

private: 
    int K;
    priority_queue<int,vector<int>,greater<>> minHeap;

public:
    KthLargest(int k, vector<int>& a) 
    {
        K=k;
        for(int ele : a)
        {
            minHeap.push(ele);
            if(minHeap.size()>k)
            minHeap.pop();
        }
    }
    
    int add(int val) 
    {
        minHeap.push(val);
        if(minHeap.size()>K)
        minHeap.pop();
        return minHeap.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 789-kth-largest-element-in-a-stream/kth-largest-element-in-a-stream.java
class KthLargest {

    int k;
    PriorityQueue<Integer> minHeap=new PriorityQueue <>();
    public KthLargest(int K,int[] nums) 
    {
        k=K;
        for(int i=0;i<nums.length;i++)
        add(nums[i]);
    }
    
    public int add(int val) 
    {
        minHeap.add(val);
        if(minHeap.size()>k)
        minHeap.poll();
        return minHeap.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */